### PR TITLE
chore(ssa): Add doc comments on `Opcode` struct

### PR DESCRIPTION
**Pull request by kevaundray**
# Related issue(s)

Resolves # 

# Description

This PR adds a doc comment on the `Opcode` struct. The word opcode is perhaps getting overused here, since ACIR also has opcodes, though this is a separate discussion.

## Test additions / changes

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

File: crates/noirc_evaluator/src/ssa/builtin.rs [hunk 1]
```diff
@@ -37,34 +46,42 @@ impl Opcode {
...
+                    // Verify returns zero or one
                     BlackBoxFunc::SchnorrVerify
                     | BlackBoxFunc::EcdsaSecp256k1
-                    | BlackBoxFunc::MerkleMembership => BigUint::one(), //verify returns 0 or 1
+                    | BlackBoxFunc::MerkleMembership => BigUint::one(),
                     BlackBoxFunc::HashToField128Security => ObjectType::NativeField.max_size(),
-                    _ => todo!("max value must be implemented for opcode {} ", op),
+                    BlackBoxFunc::RANGE
+                    | BlackBoxFunc::AES
+                    | BlackBoxFunc::AND
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa/builtin.rs:61]: Note; The benefit of removing the wildcard reveals that we are panicking on the Range, And, and Xor case

File: crates/noirc_evaluator/src/ssa/builtin.rs [hunk 2]
```diff
@@ -25,7 +34,7 @@ impl Opcode {
         }
     }
 
-    pub fn name(&self) -> &str {
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa/builtin.rs:28]: `name()` is only used for displaying, so it does not need to be public

File: crates/noirc_evaluator/src/ssa/builtin.rs [hunk 3]
```diff
@@ -37,34 +46,42 @@ impl Opcode {
...
+    /// Returns the number of elements that the `Opcode` should return
+    /// and the type.
     pub fn get_result_type(&self) -> (u32, ObjectType) {
         match self {
             Opcode::LowLevel(op) => {
                 match op {
-                    BlackBoxFunc::AES => (0, ObjectType::NotAnObject), //Not implemented
-                    BlackBoxFunc::SHA256 => (32, ObjectType::Unsigned(8)),
-                    BlackBoxFunc::Blake2s => (32, ObjectType::Unsigned(8)),
+                    BlackBoxFunc::AES => todo!("ICE: AES is unimplemented"),
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa/builtin.rs:77]: Behaviour change here: If/when AES is added into the noir stdlib, we want it to panic the compiler in the places that need to change in CI

File: crates/noirc_evaluator/src/ssa/builtin.rs (hunk 1 above)
Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa/builtin.rs:61]: This PR is blocked by this comment.

Comment (guipublic) [reply crates/noirc_evaluator/src/ssa/builtin.rs:61]: You can use an `unreachable("unexpected opcode")` here if you want.

Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa/builtin.rs:61]: Why is range, and and  xor unexpected?

**Review by guipublic — APPROVED** [review]

**Pull request merged**